File: BinaryTree/层序遍历.swift

public class TreeNode {
    public var val: Int
    public var left: TreeNode?
    public var right: TreeNode?
    public init(_ val: Int) {
        self.val = val
        self.left = nil
        self.right = nil
    }
}
class Solution {
    func levelOrder(_ root: TreeNode?) -> [[Int]] {
        var res: [[Int]] = []
        var dfs: [TreeNode?] = []
        dfs.append(root!)
        while !dfs.isEmpty {
            var _res: [Int] = []
            var _dfs: [TreeNode?] = []
            for temp in dfs {
                if temp != nil {
                    _res.append(temp!.val)
                    if temp!.left != nil {_dfs.append(temp!.left)}
                    if temp!.right != nil {_dfs.append(temp!.right)}   
                }
            }
            dfs = _dfs
            if _res.count > 0 {
                res.append(_res)
            }
        }
        return res
    }
}

File: BinaryTree/层序遍历.cpp
#include <vector>
#include <queue>
using namespace std;

struct TreeNode {
    int val;
    TreeNode *left;
    TreeNode *right;
     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
};
 
class Solution {
public:
    vector<vector<int> > res;
    void bfs(TreeNode* root) {
        if (!root) return ;
        queue<TreeNode*> bfsq;
        bfsq.push(root);
        while (!bfsq.empty()) {
            res.push_back(vector<int> ());
            for(int i = bfsq.size(); i > 0; i--) {
                TreeNode* temp = bfsq.front();
                res.back().push_back(temp->val);
                bfsq.pop();
                if (temp->left) bfsq.push(temp->left);
                if (temp->right) bfsq.push(temp->right);
            }
        }
    }
    vector<vector<int> > levelOrder(TreeNode* root) {
        bfs(root);
        return res;
    }
};